### PR TITLE
Roles with course runs.

### PR DESCRIPTION
**Pull request by awais786**
WIP: Initial db structure for roles

Created table for UserRoles.
Created relationship between Organizations and groups with 3 default roles. 

ECOM-6216

### CODE REVIEW

File: course_discovery/apps/publisher/models.py [hunk 1]
```diff
@@ -78,6 +79,37 @@ def publish(self):
...
+        )
+
+    class Meta(object):
+
+        unique_together = (
+            ('user', 'role'),
+        )
+
+
 class Course(TimeStampedModel, ChangedByMixin):
```
Comment (bderusha) [inline course_discovery/apps/publisher/models.py:113]: We also need this user_role for a `Course`

File: course_discovery/apps/publisher/models.py [hunk 2]
```diff
@@ -354,3 +387,21 @@ def __str__(self):
 
     class Meta:
         verbose_name_plural = 'UserAttributes'
+
+
+class OrganizationGroup(TimeStampedModel):
```
Comment (bderusha) [inline course_discovery/apps/publisher/models.py]: Do we need to do this this way as opposed to letting Django handle this for us?

File: course_discovery/apps/publisher/models.py (hunk 1 above)
Comment (awais786) [reply course_discovery/apps/publisher/models.py:113]: Do we need user_role for both course-run and course ?

File: course_discovery/apps/publisher/models.py (hunk 2 above)
Comment (awais786) [reply course_discovery/apps/publisher/models.py]: How django can handle this ? For org and group we need some kind of relationship.

**Comment by awais786** [comment]
@bderusha  @jareerahsan @waheedahmed I have created db structure. Please review these db changes.

File: course_discovery/apps/publisher/models.py (hunk 1 above)
Comment (awais786) [reply course_discovery/apps/publisher/models.py:113]: Because stage changes will be occur for course runs only.

**Comment by clintonb** [comment]
Are there design documents stating how roles/permissions are supposed to work? If not, can we get those before we continue with coding? I've now seen 2-3 pull requests dealing with permissions, but no documentation. This documentation will (a) make it easier to understand/implement this functionality, and (b) make it easier for reviewers to validate it.

**Comment by bderusha** [comment]
@awais786 Can we close this PR until we have a data model design document clearly stating our approach?

**Comment by bderusha** [comment]
@awais786 can we close this?

**Comment by bderusha** [comment]
@awais786 I am closing this PR.  Please reopen if you think it should remain open.

**Pull request closed without merging**